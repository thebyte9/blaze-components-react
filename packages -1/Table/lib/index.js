"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var lodash_orderby_1 = __importDefault(require("lodash.orderby"));
var react_1 = __importStar(require("react"));
var TableBody_1 = __importDefault(require("./TableBody"));
var TableHead_1 = __importDefault(require("./TableHead"));
var Table = function (_a) {
    var _b = _a.data, columns = _b.columns, rows = _b.rows, identification = _b.identification, orderBy = _b.orderBy, _c = _a.onSelect, onSelect = _c === void 0 ? function () { return ({}); } : _c, checkboxes = _a.checkboxes, placeholder = _a.placeholder;
    var asc = "asc";
    var desc = "desc";
    var hide = true;
    var formatColumns = columns.reduce(function (result, item) {
        var _a;
        return __assign({}, result, (_a = {}, _a[item] = hide, _a));
    }, {});
    var _d = react_1.useState([]), selected = _d[0], setSelected = _d[1];
    var _e = react_1.useState(formatColumns), sortColumns = _e[0], setSortColumns = _e[1];
    var _f = react_1.useState(rows), allRows = _f[0], setRows = _f[1];
    react_1.useEffect(function () {
        setRows(rows);
        setSortColumns(formatColumns);
    }, [rows, columns]);
    var handleSelected = function (_a, value, multiselect) {
        var checked = _a[0];
        if (multiselect === void 0) { multiselect = false; }
        var checkedValue = [];
        if (checked && !selected.includes(checked.value)) {
            checkedValue = selected.concat([checked.value]);
        }
        else {
            checkedValue = selected.filter(function (currentValue) { return currentValue !== value; });
        }
        if (multiselect) {
            checkedValue = checked ? checked.value.slice() : [];
        }
        updateSelected(checkedValue);
    };
    var updateSelected = function (selectedRows) {
        setSelected(selectedRows);
        onSelect(selectedRows);
    };
    var getSortDirection = function (column) {
        if (sortColumns[column] === hide) {
            return asc;
        }
        return sortColumns[column] === asc ? desc : asc;
    };
    var sort = function (column) {
        var _a;
        if (!orderBy.includes(column)) {
            return;
        }
        var resetSortColumns = {};
        Object.keys(sortColumns).forEach(function (key) { return (resetSortColumns[key] = hide); });
        var sortDirection = getSortDirection(column);
        setRows(lodash_orderby_1["default"](allRows, [column], [sortDirection]).slice());
        setSortColumns(__assign({}, resetSortColumns, (_a = {}, _a[column] = sortDirection, _a)));
    };
    var enableOrderBy = function (column) { return (react_1["default"].createElement(react_1.Fragment, null,
        react_1["default"].createElement("span", { "data-testid": "sortby-" + column, onClick: function () { return sort(column); }, role: "button" }, column),
        sortColumns[column] !== hide && (react_1["default"].createElement("i", { className: "material-icons" }, sortColumns[column] === asc
            ? "keyboard_arrow_up"
            : "keyboard_arrow_down")))); };
    return (react_1["default"].createElement("div", { className: "table-scroll__wrapper" },
        react_1["default"].createElement("div", { className: "table-scroll" },
            react_1["default"].createElement("table", null,
                react_1["default"].createElement(TableHead_1["default"], { checkboxes: checkboxes, selected: selected, identification: identification, allRows: allRows, handleSelected: handleSelected, enableOrderBy: enableOrderBy, sortColumns: sortColumns }),
                react_1["default"].createElement(TableBody_1["default"], { allRows: allRows, checkboxes: checkboxes, identification: identification, selected: selected, handleSelected: handleSelected, columns: columns, placeholder: placeholder })))));
};
Table.defaultProps = {
    checkboxes: false,
    data: {
        columns: [],
        identification: "",
        orderBy: [],
        rows: []
    },
    placeholder: "No records available"
};
exports["default"] = Table;
