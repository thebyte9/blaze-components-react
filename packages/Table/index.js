import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import uuidv1 from 'uuid/v1';
import Checkboxes from '@blaze-react/checkboxes';

const Table = ({ data: { columns, rows, identification }, onSelect, checkboxes, placeholder }) => {
  const [selected, setSelected] = useState([]);

  const handleSelected = ([checked], value, multiselect = false) => {
    if (multiselect) {
      if (checked) setSelected([...checked.value]);
      else setSelected([]);
      return;
    }

    if (checked && !selected.includes(checked.value)) setSelected([...selected, checked.value]);
    else setSelected(selected.filter(_value => _value !== value));
  };

  useEffect(() => onSelect(selected));

  const colSpan = checkboxes ? columns.length + 1 : columns.length;

  const thead = (
    <thead>
      <tr>
        {checkboxes && (
          <th>
            <Checkboxes
              withEffect
              options={[
                Object.assign(
                  {},
                  {
                    value: rows.map(row => row[identification]),
                    id: 'Select_all',
                    checked: selected.length === rows.length
                  }
                )
              ]}
              onChange={({ checked }) => handleSelected(checked, checked, true)}
            />
          </th>
        )}
        {columns.map(column => <th key={uuidv1()}>{column}</th>)}
      </tr>
    </thead>
  );

  const tbody = (
    <tbody>
      {rows.map(row => (
        <tr key={uuidv1()}>
          {checkboxes && (
            <td>
              <Checkboxes
                withEffect
                options={[
                  {
                    value: row[identification],
                    id: row[identification],
                    checked: selected.includes(row[identification])
                  }
                ]}
                onChange={({ checked }) => handleSelected(checked, row[identification])}
              />
            </td>
          )}
          {columns.map(column => <td key={uuidv1()}>{row[column]}</td>)}
        </tr>
      ))}
      {!rows.length && (
        <tr>
          <td colSpan={colSpan} align="center">
            {placeholder}
          </td>
        </tr>
      )}
    </tbody>
  );

  return (
    <div className="table-scroll__wrapper">
      <div className="table-scroll">
        <table>
          {thead}
          {tbody}
        </table>
      </div>
    </div>
  );
};

Table.propTypes = {
  placeholder: PropTypes.string,
  checkboxes: PropTypes.bool,
  data: PropTypes.shape({
    identification: PropTypes.string,
    columns: PropTypes.arrayOf(PropTypes.string),
    rows: PropTypes.array
  }),
  onSelect: PropTypes.func
};

Table.defaultProps = {
  placeholder: 'No records available',
  checkboxes: false,
  data: {},
  onSelect: () => {}
};

export default Table;
