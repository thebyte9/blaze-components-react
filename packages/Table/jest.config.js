module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'jsdom',
  globals: {
    'ts-jest': {
      isolatedModules: true,
    },
  },
  testPathIgnorePatterns: ['/node_modules/', '/mocks/'],
};
