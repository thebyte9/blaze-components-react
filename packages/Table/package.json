{
    "name": "@blaze-react/table",
    "version": "0.2.3",
    "publishConfig": {
        "access": "public"
    },
    "description": "",
    "main": "./lib/index.js",
    "scripts": {
        "build": "../../node_modules/.bin/babel ./src --out-dir lib ",
        "build-storybook": "build-storybook -c .storybook -o docs-build",
        "lint": "eslint ./ --ignore-path ../../.eslintignore",
        "test": "jest",
        "test:coverage": "jest --coverage",
        "pre-publish": "npm run build && npm run lint && npm run test"
    },
    "keywords": [],
    "author": "",
    "license": "ISC",
    "dependencies": {
        "@blaze-react/checkboxes": "^0.2.3",
        "@types/uuid": "^3.4.4"
    },
    "devDependencies": {
        "@babel/cli": "^7.4.4",
        "@babel/core": "^7.4.5"
    },
    "peerDependencies": {
        "prop-types": "^15.7.2",
        "react": "^16.8.6",
        "react-dom": "^16.8.6"
    },
    "babel": {
        "presets": [
            "@babel/preset-env",
            "@babel/preset-react"
        ]
    },
    "gitHead": "1edd29612d1854f0418c0b81d8755dd0cbd8bfb5"
}