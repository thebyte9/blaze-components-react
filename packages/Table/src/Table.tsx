import React, { FunctionComponent, useEffect, useRef, useState } from "react";
import TableBody from "./TableBody";
import TableHead from "./TableHead";
import { ITableRow, ICheckboxItem } from './interfaces';

interface ITableProps {
  placeholder?: string;
  checkboxes?: boolean;
  data: {
    identification: string;
    columns: string[];
    orderBy: string[];
    rows: ITableRow[];
    appliedSort?: any;
    labels: Record<string, unknown>;
  };
  value?: string;
  overScanBuffer?: number;
  onSelect?: (arg: any[]) => any;
  onSort?: (arg: any) => any;
  onRenderItems?: (arg: any) => void;
  onClickRow?: (arg: any) => void;
  scrollToIndex?: number;
}


const Table: FunctionComponent<ITableProps> = ({
  data: { columns, rows, identification, orderBy, appliedSort, labels },
  onSelect = () => ({}),
  onSort = () => ({}),
  onClickRow = () => ({}),
  checkboxes,
  placeholder = '', 
  overScanBuffer = 0,
  onRenderItems,
  scrollToIndex = 0,
}) => {
  const [selected, setSelected] = useState<any[]>([]);
  const [allRows, setAllRows] = useState<ITableRow[]>(rows);
  const [allColumns, setAllColumns] = useState<any>(columns);
  const headRef = useRef<any>(null);
  const bodyRef = useRef<any>(null);

  useEffect(() => {
    setAllRows(rows);
    setAllColumns(columns);
  }, [rows, rows && rows.length, columns, columns && columns.length]);

  useEffect(() => {
    if (
      bodyRef &&
      bodyRef.current &&
      bodyRef.current.firstElementChild &&
      allRows.length
    ) {
      bodyRef.current.firstElementChild.addEventListener(
        "scroll",
        (event: any) => syncScroll(headRef.current, event)
      );
    }

    if (headRef && headRef.current && headRef.current.firstElementChild) {
      headRef.current.addEventListener("scroll", (event: any) =>
        syncScroll(bodyRef.current.firstElementChild, event)
      );
    }
    return () => {
      if (bodyRef && bodyRef.current && bodyRef.current.firstElementChild) {
        bodyRef.current.firstElementChild.removeEventListener(
          "scroll",
          syncScroll
        );
      }
      if (headRef.current) {
        headRef.current.removeEventListener("scroll", syncScroll);
      }
    };
  }, [bodyRef.current, headRef.current, allRows]);

  const syncScroll = (ref: any, event: any) => {
    ref.scrollLeft = event.target.scrollLeft;
  };

  const handleSelected = (
    [checked]: ICheckboxItem[],
    value: string | ICheckboxItem[],
    multiselect = false
  ): void => {
    let checkedValue = [];

    if (checked && !selected.includes(checked.value)) {
      checkedValue = [...selected, checked.value];
    } else {
      checkedValue = selected.filter((currentValue) => currentValue !== value);
    }

    if (multiselect) {
      checkedValue = checked ? [...checked.value] : [];
    }

    setSelected(checkedValue);
    onSelect(checkedValue);
  };

  return (
    <div className="table-wrapper">
      <TableHead
        onSort={onSort}
        orderBy={orderBy}
        headRef={headRef}
        columns={allColumns}
        appliedSort={appliedSort}
        labels={labels}
      />
      <TableBody
        scrollToIndex={scrollToIndex}
        onClickRow={onClickRow}
        bodyRef={bodyRef}
        allRows={allRows}
        checkboxes={checkboxes}
        identification={identification}
        selected={selected}
        handleSelected={handleSelected}
        columns={columns}
        placeholder={placeholder}
        overScanBuffer={overScanBuffer}
        onRenderItems={onRenderItems}
      />
    </div>
  );
};

Table.defaultProps = {
  checkboxes: false,
  data: {
    columns: [],
    identification: "",
    orderBy: [],
    rows: [],
    labels: {},
  },
  placeholder: "No records available",
};
export default Table;
