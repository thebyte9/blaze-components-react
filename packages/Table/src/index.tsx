import React, { FunctionComponent, useEffect, useState } from "react";
import Checkboxes from "@blaze-react/checkboxes";
import utils from '@blaze-react/utils';
import _orderBy from "lodash.orderby";

interface ITableProps {
  placeholder?: string;
  checkboxes?: boolean;
  utils: {
    uniqueId: Function
  };
  data: {
    identification: string;
    columns: string[];
    orderBy: string[];
    rows: any[];
  };
  value?: string;
  onSelect: (...args: any[]) => any;
}
const Table: FunctionComponent<ITableProps> = ({
  data: { columns, rows, identification, orderBy },
  onSelect,
  checkboxes,
  placeholder,
  utils: { uniqueId },
  value
}) => {

  const [selected, setSelected] = useState<any[]>([]);
  const [allRows, setRows] = useState(rows);
  const [sortColumns, setSortColumns] = useState(
    columns.reduce((result, item) => ({ ...result, [item]: "asc" }), {})
  );

  const handleSelected = (
    [checked]: [{ checked: boolean; id: string | number; value: any }],
    value: String,
    multiselect = false
  ) => {
    if (multiselect) {
      let checkedValue = [];
      if (checked) {
        checkedValue = Array.isArray(checked.value)
          ? [...checked.value]
          : [checked.value];
      }
      setSelected(checkedValue);
      return;
    }
    if (checked && !selected.includes(checked.value)) {
      setSelected([...selected, checked.value]);
    } else {
      setSelected(selected.filter(currentValue => currentValue !== value));
    }
  };

  const sort = (column: any) => {
    setRows([..._orderBy(allRows, [column], [sortColumns[column]])]);

    setSortColumns({
      ...sortColumns,
      [column]: sortColumns[column] === "asc" ? "desc" : "asc"
    });
  };

  useEffect(() => onSelect(selected));
  const colSpan = checkboxes ? columns.length + 1 : columns.length;

  const enableOrderBy = (column: string) => orderBy.includes(column) ? (
    <i
      id={`sort_${column}`}
      className="material-icons"
      onClick={() => sort(column)}
      role="button"
    >
      {sortColumns[column] === "asc"
        ? "keyboard_arrow_up"
        : "keyboard_arrow_down"}
    </i>
  ) : null;

  const thead = (
    <thead>
      <tr>
        {checkboxes && (
          <th>
            <Checkboxes
              withEffect
              options={[
                Object.assign(
                  {},
                  {
                    checked: selected.length === allRows.length,
                    id: "Select_all",
                    value: allRows.map(row => row[identification])
                  }
                )
              ]}
              onChange={({ checked }: { checked: any }) =>
                handleSelected(checked, checked, true)
              }
            />
          </th>
        )}
        {Object.keys(sortColumns).map(column => (
          <th key={uniqueId(column)}>
            {column}
            {enableOrderBy(column)}
          </th>
        ))}
      </tr>
    </thead>
  );
  const tbody = (
    <tbody>
      {allRows.map(row => (
        <tr key={uniqueId(row)}>
          {checkboxes && (
            <td>
              <Checkboxes
                withEffect
                options={[
                  {
                    checked: selected.includes(row[identification]),
                    id: row[identification],
                    value: row[identification]
                  }
                ]}
                onChange={({ checked }: any) =>
                  handleSelected(checked, row[identification])
                }
              />
            </td>
          )}
          {columns.map(column => (
            <td key={column}>{row[column]}</td>
          ))}
        </tr>
      ))}
      {!allRows.length && (
        <tr>
          <td colSpan={colSpan} align="center">
            {placeholder}
          </td>
        </tr>
      )}
    </tbody>
  );
  return (
    <div className="table-scroll__wrapper">
      <div className="table-scroll">
        <table>
          {thead}
          {tbody}
        </table>
      </div>
    </div>
  );
};

Table.defaultProps = {
  checkboxes: false,
  data: {
    columns: [],
    orderBy: [],
    identification: "",
    rows: []
  },
  onSelect: (): void => {
    return;
  },
  placeholder: "No records available"
};
export default utils(Table);
