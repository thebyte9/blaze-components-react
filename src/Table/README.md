
## Description

The HTML table element represents tabular data â€” that is, information presented in a two-dimensional table comprised of rows and columns of cells containing data.

## Usage

* With Checkboxes

```js
const data =  {
  identification: 'id',
  columns: ['name', 'age'], 
  rows: [{
    id: 1,
    name: 'Oscar Leon',
    age: 26,
  }, {
    id: 2,
    name: 'Ismael Haytam',
    age: 23,
  }, {
    id: 3,
    name: 'Lorem Ipsum',
    age: 45
  }]
}

<Table 
  checkboxes 
  data={data} 
  onSelect={(selected) => {}} />
```

* Static table

```js
<Table data={data} />
```

## API

##### Table can receive a number of `props` as follow:


| NAME   | TYPE | DEFAULT | 
| :---  | :---:  | :---: | 
| placeholder | string| No records available | 
| data | Object | {}      | 
| onSelect | Function | () => {}  | 
| checkboxes | Boolean | false | 


