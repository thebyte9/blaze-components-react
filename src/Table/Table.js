import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import uuidv1 from 'uuid/v1';
import sortBy from 'sort-by';
import Checkboxes from '../Checkboxes';

const Table = ({
  data: {
    columns,
    rows,
    identification,
  },
  onSelect,
  checkboxes,
  placeholder,
}) => {
  const [_rows, setRows] = useState(rows);
  const [selected, setSelected] = useState([]);

  const [sortColumns, setSortColumns] = useState(
    columns.reduce((result, item) => ({ ...result, [item]: item }), {})
  );

  const handleSelected = ([checked], value, multiselect = false) => {
    if (multiselect) {
      if (checked) setSelected([...checked.value]);
      else setSelected([]);
      return;
    }

    if (checked && !selected.includes(checked.value)) setSelected([...selected, checked.value]);
    else setSelected(selected.filter(_value => _value !== value));
  };

  const sort = (column) => {
    setRows([..._rows.sort(sortBy(sortColumns[column]))]);

    setSortColumns({
      ...sortColumns,
      [column]: (sortColumns[column] === column) ? `-${column}` : column
    });
  };

  useEffect(() => onSelect(selected));

  const colSpan = checkboxes ? columns.length + 1 : columns.length;

  const thead = (
    <thead>
      <tr>
        {
          checkboxes && (
            <th>
              <Checkboxes
                withEffect
                options={[
                  Object.assign(
                    {},
                    {
                      value: _rows.map(row => row[identification]),
                      id: 'Select_all',
                      checked: selected.length === _rows.length
                    }
                  )
                ]}
                onChange={({ checked }) => handleSelected(checked, checked, true)}
                />
            </th>
          )
        }
        {Object.keys(sortColumns).map(column => (
          <th key={uuidv1()}>
            {column}
            <i id={`sort_${column}`} className="material-icons" onClick={() => sort(column)} role="button">
              {sortColumns[column].includes('-') ? 'keyboard_arrow_up' : 'keyboard_arrow_down'}
            </i>
          </th>
        ))}
      </tr>
    </thead>
  );

  const tbody = (
    <tbody>
      {
        _rows.map(row => (
          <tr key={uuidv1()}>
            {
              checkboxes
              && (
                <td>
                  <Checkboxes
                  withEffect
                  options={
                    [
                      {
                        value: row[identification],
                        id: row[identification],
                        checked: selected.includes(row[identification])
                      }
                    ]
                  }
                  onChange={({ checked }) => handleSelected(checked, row[identification])}
                  />
                </td>
              )
            }
            {columns.map(column => <td key={uuidv1()}>{row[column]}</td>)}
          </tr>
        ))
      }
      {!_rows.length && (
      <tr>
        <td colSpan={colSpan} align="center">{placeholder}</td>
      </tr>
      )}
    </tbody>
  );

  return (
    <div className="table-scroll__wrapper">
      <div className="table-scroll">
        <table>
          {thead}
          {tbody}
        </table>
      </div>
    </div>
  );
};

Table.propTypes = {
  placeholder: PropTypes.string,
  checkboxes: PropTypes.bool,
  data: PropTypes.shape({
    identification: PropTypes.string,
    columns: PropTypes.arrayOf(PropTypes.string),
    rows: PropTypes.array
  }),
  onSelect: PropTypes.func
};

Table.defaultProps = {
  placeholder: 'No records available',
  checkboxes: false,
  data: {},
  onSelect: () => {}
};

export default Table;
