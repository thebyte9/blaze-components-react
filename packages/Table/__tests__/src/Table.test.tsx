import '@testing-library/jest-dom';

import React from 'react';
import Table from '../../src/Table';
import { data } from '../mocks';
import { render } from '@testing-library/react';

const paginationProps = {
  display: true,
  paginationPagesPerSide: 5,
  itemsPerPage: 10,
  currentPage: 1
}

describe('Table component', () => {
  it('should be defined and renders correctly (snapshot)', () => {
    const { asFragment } = render(<Table data={data} checkboxes={false} paginationProps={paginationProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should run the cleanup function inside useEffect', () => {
    const useEffectSpy = jest.spyOn(React, 'useEffect');

    const { rerender } = render(<Table data={data} checkboxes paginationProps={paginationProps} />);
    expect(useEffectSpy).toHaveBeenCalled();

    rerender(<Table data={data} checkboxes paginationProps={paginationProps} />);
  });
});
