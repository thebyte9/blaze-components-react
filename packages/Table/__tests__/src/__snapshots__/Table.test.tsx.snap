// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should be defined and renders correctly (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="table-wrapper"
  >
    <div
      class="table-head"
    >
      <div
        class="table-cell--checkbox"
      />
      <div
        class="table-cell"
      >
        <div
          class="sortable"
        >
          <span
            data-testid="sortby-id"
            role="button"
          >
            id
          </span>
        </div>
      </div>
      <div
        class="table-cell"
      >
        <div
          class="sortable"
        >
          <span
            data-testid="sortby-name"
            role="button"
          >
            Name
          </span>
        </div>
      </div>
      <div
        class="table-cell"
      >
        <div
          class="sortable"
        >
          <span
            data-testid="sortby-age"
            role="button"
          >
            age
          </span>
        </div>
      </div>
    </div>
    <div
      class="table-body"
    >
      <div
        class="custom-virtual-list"
      >
        <div
          class="table-row"
          data-testid="tablerow-0"
          style="height: 62px;"
        >
          <div
            class="table-cell"
          >
            <div
              class="table-cell--content"
            >
              1
            </div>
          </div>
          <div
            class="table-cell"
          >
            <div
              class="table-cell--content"
            >
              Lorem
            </div>
          </div>
          <div
            class="table-cell"
          >
            <div
              class="table-cell--content"
            >
              52
            </div>
          </div>
        </div>
        <div
          class="table-row"
          data-testid="tablerow-1"
          style="height: 62px;"
        >
          <div
            class="table-cell"
          >
            <div
              class="table-cell--content"
            >
              2
            </div>
          </div>
          <div
            class="table-cell"
          >
            <div
              class="table-cell--content"
            >
              Ipsum
            </div>
          </div>
          <div
            class="table-cell"
          >
            <div
              class="table-cell--content"
            >
              43
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
